### PR TITLE
Support ANSI escape sequences in `Kino.Text`

### LINKED ISSUE #313: Add ANSI escape support to `Kino.Text`
Currently, `Kino.Text` always renders as `Kino.Output.plain_text(content)`. To render text containing ANSI escape sequences, you have to create your own wrapper struct that uses `Kino.Output.text(content)`.

Being able to use `Kino.Text` to support text with ANSI escape sequences would be a nice little improvement:

```elixir
Kino.Text.new(my_text, plain: false) # default true
# or
Kino.Text.new(my_text, ansi: true) # default false
```

### PR DESCRIPTION
**Pull request by zachallaun**
Closes #313

### CODE REVIEW

File: lib/kino/text.ex [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+
+    * `:console` - whether to render the text as if it were printed to
+      standard output, supporting ANSI escape codes. Defaults to `false`
   """
-  @spec new(String.t()) :: t()
-  def new(content) do
-    %__MODULE__{content: content}
+  @spec new(String.t(), opts) :: t() when opts: [console: boolean()]
+  def new(content, opts \\ []) when is_binary(content) do
+    %__MODULE__{content: content, console: Keyword.get(opts, :console, false)}
```
Comment (jonatanklosko) [inline lib/kino/text.ex:37]:
```suggestion
    opts = Keyword.validate!(opts, console: false)
    %__MODULE__{content: content, console: opts[:console]}
```

**Review by jonatanklosko — APPROVED** [review]

File: lib/kino/text.ex [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
+            content: binary(),
+            console: boolean()
           }
 
   @doc """
   Creates a new kino displaying the given text content.
+
+  ## Options
+
+    * `:console` - whether to render the text as if it were printed to
```
Comment (jonatanklosko) [inline lib/kino/text.ex:32]: Oh actually I think `:terminal` is a better word.

File: test/kino/text_test.exs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    test "outputs console text" do
+      [:red, "Hello!"]
+      |> IO.ANSI.format()
+      |> IO.iodata_to_binary()
+      |> Kino.Text.new(console: true)
+      |> Kino.render()
+
+      assert_output({:text, "\e[31mHello!\e[0m"})
```
Comment (jonatanklosko) [inline test/kino/text_test.exs:20]: On CI Elixir disables ANSI. We could enable it explicitly, but I don't think it's relevant for the test, the content will be whatever we pass, just need to make sure it's the right output type :)

```suggestion
      "Hello!" |> Kino.Text.new(console: true) |> Kino.render()
      assert_output({:text, "Hello!"})
```

**Comment by zachallaun** [comment]
All requested changes made :)

**Comment by jonatanklosko** [comment]
🐈!

**Pull request merged**